stdlib/public/SDK/Foundation/TypePreservingNSNumber.mm

```diff
@@ -23,26 +25,50 @@ @interface _SwiftTypePreservingNSNumber : NSNumber
 
 // This enum has a matching counterpart in Foundation.swift, please
 // update both copies when changing it.
-enum _SwiftTypePreservingNSNumberTag {
-  SwiftInt = 1,
-  SwiftUInt = 2,
-  SwiftFloat = 3,
-  SwiftDouble = 4,
-  SwiftCGFloat = 5,
-  SwiftBool = 6
+enum _SwiftTypePreservingNSNumberTag: uint8_t {
+  SwiftInt     =  0,
+  SwiftInt64   =  1,
+  SwiftInt32   =  2,
+  SwiftInt16   =  3,
+  SwiftInt8    =  4,
+  SwiftUInt    =  5,
+  SwiftUInt64  =  6,
+  SwiftUInt32  =  7,
+  SwiftUInt16  =  8,
+  SwiftUInt8   =  9,
+  SwiftFloat   = 10,
+  SwiftDouble  = 11,
+  SwiftCGFloat = 12,
+  SwiftBool    = 13,
+  /// A non-type-preserving NSNumber from Cocoa.
+  ///
+  /// Note that this value is *not* in the corresponding Swift enum in
+  /// Foundation.swift. It should never be used as the tag of a valid
+  /// _SwiftTypePreservingNSNumber instance; it is used as a return value
+  /// for _swift_Foundation_TypePreservingNSNumberGetKind to indicate that
+  /// an NSNumber instance does not know its originating Swift type.
+  ///
+  /// The numbering scheme here, with `NonSwift` coming after the valid values
+  /// for the enum, matches Swift's tag assignment scheme for single-payload
+  /// enums. For an Optional<_SwiftTypePreservingNSNumberTag>, Swift uses the
+  /// first invalid integer value to represent `.none`. This representation on
+  /// the C side allows _SwiftTypePreservingNSNumberTag(rawValue:) on the Swift
+  /// side to compile down to a no-op.
```

Comment:
That's how it was before, but that then forces a representation change when we convert from the raw C value to the Swift enum on the Swift side, since the Swift enum always uses the compacted representation starting from zero. Definitely a microoptimization.